lib/HaskellLib_H.thy

```diff
@@ -548,6 +548,30 @@ lemma gets_the_ohaskell_assert:
   "gets_the (ohaskell_assert P []) = assert P"
   by (clarsimp simp: ohaskell_assert_def split: if_splits)
 
-lemmas omonad_defs = omonad_defs ohaskell_assert_def ohaskell_fail_def
+definition ostate_assert :: "('s \<Rightarrow> bool) \<Rightarrow> ('s, unit) lookup" where
+  "ostate_assert P \<equiv> do { s \<leftarrow> ask; oassert (P s)}"
```

Comment:
This definition should be in the monad library, only `ohaskell_state_assert` should live on HaskellLib_H